src/ext/php7/handlers_mysqli.c

```diff
@@ -0,0 +1,46 @@
+#include "handlers_internal.h"
+
+// clang-format off
+static ddtrace_string ddtrace_mysqli_functions[] = {
+    DDTRACE_STRING_LITERAL("mysqli_commit"),
+    DDTRACE_STRING_LITERAL("mysqli_connect"),
+    DDTRACE_STRING_LITERAL("mysqli_prepare"),
+    DDTRACE_STRING_LITERAL("mysqli_query"),
+    DDTRACE_STRING_LITERAL("mysqli_real_connect"),
+    DDTRACE_STRING_LITERAL("mysqli_stmt_execute"),
+    DDTRACE_STRING_LITERAL("mysqli_stmt_get_result"),
+};
+
+static ddtrace_string ddtrace_mysqli_methods[] = {
+    DDTRACE_STRING_LITERAL("__construct"),
+    DDTRACE_STRING_LITERAL("commit"),
+    DDTRACE_STRING_LITERAL("prepare"),
+    DDTRACE_STRING_LITERAL("query"),
+    DDTRACE_STRING_LITERAL("real_connect"),
+};
+
+static ddtrace_string ddtrace_mysqli_stmt_methods[] = {
+    DDTRACE_STRING_LITERAL("execute"),
+    DDTRACE_STRING_LITERAL("get_result"),
+};
+// clang-format on
+
+void ddtrace_mysqli_handlers_startup(void) {
+    // todo: how to handle ddtrace_resource = -1?
```

Comment:
Is this something that needs to be addressed before merge you think?

Reply:
I think so; thanks for bringing it up. Prior to this change we lose an optimization. With this change we lose core functionality if we can't get a resource. If this happens we need to:

- Do it safely and make sure we don't access `reserved[-1]` which would sigsegv.
- Emit a message saying that we aren't tracing internal functions and to open a report. Not sure what info we can log; I don't think you can reverse-map the numbers to extensions that asked for them.